docs: add basic and advanced documentation and refactor main readme (#72)
microsoft/data-factory-testing-framework: docs/basic/repository_setup.md

# Repository setup

## Git integration

> To be able to write tests for data factory, you need to have the pipeline and activity definitions available. The recommended way to do this is to sync the Data Factory instance to a git repository, so that you can create a `tests` folder in the same repository and write tests for your data factory. The git integration process can be found here:

Reviewer: ```suggestion
To be able to write tests for data factory, the pipeline and activity definitions need to be available. The recommended way to do this is to sync the Data Factory instance to a git repository, so that a `tests` folder can be created in the same repository and write tests for your data factory. The git integration process can be found here:
```

> 1. [Fabric - Git integration process](https://learn.microsoft.com/en-us/fabric/cicd/git-integration/git-integration-process)

Reviewer: Remove localization from the link
```suggestion
1. [Fabric - Git integration process](https://learn.microsoft.com/fabric/cicd/git-integration/git-integration-process)
```

2. [Azure Data Factory - Git integration process](https://learn.microsoft.com/azure/data-factory/source-control)

### Alternative for Azure Data Factory

> If you want to download a single JSON file for testing purposes, you can do so by following these steps:

Reviewer: ```suggestion
To download a single JSON file for testing purposes,  follow these steps:
```

> 1. Open your Data Factory instance, and open the pipeline you want to test.

Reviewer: ```suggestion
1. Open the Data Factory instance, and open the pipeline to be tested.
```

2. Click on the action ellipses
3. Click "Download support files"

> 4. Extract the zip file containing the pipeline definition in a folder of your choice.

Reviewer: ```suggestion
4. Extract the zip file containing the pipeline definition in a folder of choice.
```

> > Remember the location of this folder, as you will need it to initialize the framework.

Reviewer: ```suggestion
> Remember the location of this folder, as it will be needed to initialize the framework.
```

> ![image](../images/download_support_files.png)

Reviewer 2: Either add a descriptive alternative text (compared to [image]) or remove the screenshot entirely. Reasons:
1. screenshots can be outdated and confuse users
2. are not supported for screenreaders (e.g., accessbility)
3. the description above is more than concise enough IMHO!

Author: Added description

> Once your repository is set up, you can install and initialize the framework as described in the [installing and initializing the framework](installing_and_initializing_framework.md) page.

Reviewer: ```suggestion
Once the repository is set up, the framework can be installed and initialized as described in the [installing and initializing the framework](installing_and_initializing_framework.md) page.
```
